library/lua/dfhack.lua

```diff
@@ -380,15 +380,33 @@ end
 
 local internal = dfhack.internal
 
-internal.scripts = internal.scripts or {}
-internal.scriptPath = internal.scriptPath or {}
-internal.scriptMtime = internal.scriptMtime or {}
-internal.scriptRun = internal.scriptRun or {}
+Script = defclass(Script)
+function Script:init(path)
+    self.path = path
+    self.mtime = dfhack.filesystem.mtime(path)
+    self._flags = {}
+end
+function Script:get_flags()
+    if self.flags_mtime ~= dfhack.filesystem.mtime(self.path) then
```

Comment:
Does this correctly deal with the case where the user loads or unloads a save which overrides a DFHack script with one modified at the same time? If their filesystem is confused and they installed a mod and DF and DFHack all at once all the files might have the same mtime.